lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -827,6 +848,12 @@ class SemanticsObject {
   DomElement? _childContainerElement;
 
   /// The parent of this semantics object.
+  ///
+  /// This value is not final until the tree is finalized. It is not safe to
+  /// rely on this value in the middle of a semantics tree update. It is safe to
+  /// use this value in post-update callback (see
+  /// [EngineSemanticsOwner.addOneTimePostUpdateCallback]).
+  SemanticsObject? get parent => _parent;
```

Comment:
Would be nice to have an assert that throws if we are in the middle of an update. Not sure what's the best way to do that though. Maybe a flag `EngineSemanticsOwner.isUpdating` and then:
```suggestion
  SemanticsObject? get parent {
    assert(!owner.isUpdating);
    return _parent;
  }
```

Reply:
Done.